src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Reload/ProjectReloadManager.cs

```diff
@@ -0,0 +1,560 @@
+                        }
+
+                        ReloadDelayScheduler.ScheduleAsyncTask(async (ct) => 
+                        {
+                            // Grab the UI thread so that we block until the reload of this set of 
+                            // projects completes.
+                            await _threadHandling.SwitchToUIThread();
+
+                            if(ct.IsCancellationRequested)
+                            {
+                                return;
+                            }
+
+                            // Get the list of projects and create a new empty list to put new requests
+                            List<IReloadableProject> changedProjects;
+                            lock(_changedProjects)
+                            {
+                                changedProjects = _changedProjects;
+                                _changedProjects = new List<IReloadableProject>();
+                            }
+                               
+                            var failedProjects = new List<Tuple<IReloadableProject, ProjectReloadResult>>();
+                            _threadHandling.ExecuteSynchronously(async () =>
+                            {
+                                foreach(var project in changedProjects)
+                                {
+                                    ProjectReloadResult result =  await project.ReloadProjectAsync().ConfigureAwait(true);
+
+                                    if(result == ProjectReloadResult.ReloadFailed || result == ProjectReloadResult.ReloadFailedProjectDirty)
+                                    {
+                                        failedProjects.Add(new Tuple<IReloadableProject, ProjectReloadResult>(project, result));
+                                    }
+                                }
+                            });
+
+                            ProcessProjectReloadFailures(failedProjects);
+
+                        });
+                    }
+                }
+            }
+            return VSConstants.S_OK;
+        }
+
+        /// <summary>
+        /// Prompts once for all the projects 
+        /// For each dirty project it prompts first to see what the user action needs to be and then
+        /// reloads the project if desired. Other options are do nothing amd save as and then reload
+        /// </summary>
+        private void ProcessProjectReloadFailures(List<Tuple<IReloadableProject, ProjectReloadResult>> failedProjects)
+        {
+            if(failedProjects.Count == 0)
+            {
+                return;
+            }
+
+            // Process each project. if the project is dirty in memory we need to prompt for each action. Non-dirty ones are
+            // simply reloaded
+            bool ignoreAll = false;
+            bool reloadAll = false;
+            foreach(var failure in failedProjects)
+            {
+                if(failure.Item2 == ProjectReloadResult.ReloadFailedProjectDirty)
```

Comment:
We should sync with the new changes that the solution made, with Reload Solution.